File: java/src/main/java/gov/doc/ntia/sigmf/ext/annotation/location/LocationAnnotation.java
package gov.doc.ntia.sigmf.ext.annotation.location;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import gov.doc.ntia.sigmf.Annotation;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class LocationAnnotation extends Annotation {

    @JsonProperty(value="x", required = false)
    protected Double x;

    @JsonProperty(value="y", required = false)
    protected Double y;

    @JsonProperty(value="z", required = false)
    protected Double z;

    @JsonProperty(value="speed", required = false)
    protected Double speed;

    @JsonProperty(value="bearing", required = false)
    protected Double bearing;

    public Double getX() {
        return x;
    }

    public void setX(Double x) {
        this.x = x;
    }

    public Double getY() {
        return y;
    }

    public void setY(Double y) {
        this.y = y;
    }

    public Double getZ() {
        return z;
    }

    public void setZ(Double z) {
        this.z = z;
    }

    public Double getSpeed() {
        return speed;
    }

    public void setSpeed(Double speed) {
        this.speed = speed;
    }

    public Double getBearing() {
        return bearing;
    }

    public void setBearing(Double bearing) {
        this.bearing = bearing;
    }




}

File: Csharp/sigmf/sigmf/ext/Annotationdir/location/LocationAnnotation.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    public class LocationAnnotation : Annotation
    {
        [JsonProperty("x")]
        protected double x;

        [JsonProperty("y")]
        protected double y;

        [JsonProperty("z")]
        protected double z;

        [JsonProperty("speed")]
        protected double speed;

        [JsonProperty("bearing")]
        protected double bearing;

        public double X { get => x; set => x = value; }
        public double Y { get => y; set => y = value; }
        public double Z { get => z; set => z = value; }
        public double Speed { get => speed; set => speed = value; }
        public double Bearing { get => bearing; set => bearing = value; }
    }
}
